### PR TITLE
Try to fix main

### PR DESCRIPTION
**Pull request by RubenRBS**
The recent TS PRs #4105 #4132 #4143 had some issues still. This PR aims to fix all I can find. As a PR now to be able to disucuss them before I'm done checking all 3 PRs

### CODE REVIEW

File: static/panes/compiler.ts [hunk 1]
```diff
@@ -1453,11 +1409,14 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
             if (result.languageId) {
                 monaco.editor.setModelLanguage(editorModel, result.languageId);
             } else {
-                const monacoDisassembly =
-                    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-                    (languages[this.currentLangId ?? '']
-                        ? languages[this.currentLangId ?? ''].monacoDisassembly
-                        : null) || 'asm';
+                let monacoDisassembly = 'asm';
+                if (this.currentLangId && this.currentLangId in languages) {
```
Comment (RubenRBS) [inline static/panes/compiler.ts:1413]: Probably not needed, compiler has to have a language. Only case it won't work is in dev if you open a state that has a compiler opened with a language that you disabled between reloads

**Review by RubenRBS — COMMENTED** [review]
Lots more to do, but I need to log off for the day

File: static/panes/compiler.ts [hunk 2]
```diff
@@ -2881,8 +2827,7 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
         const are = unwiseOptions.length === 1 ? ' is ' : ' are ';
         const msg = options + names + are + 'not recommended, as behaviour might change based on server hardware.';
 
-        // @ts-ignore
-        if (optionsArray.some(opt => opt === '-flto') && !this.filters.state.binary && !wasCmake) {
+        if (optionsArray.some(opt => opt === '-flto') && !this.filters.isSet('binary') && !wasCmake) {
```
Comment (partouf) [inline static/panes/compiler.ts:2830]: is this a smart thing to do? this wouldn't detect typos for filternames

**Comment by partouf** [comment]
be aware I have a bunch of new types regarding `resultline` in https://github.com/compiler-explorer/compiler-explorer/pull/4139/files so don't spend too much time with that

File: static/panes/compiler.ts (hunk 2 above)
Comment (mattgodbolt) [reply static/panes/compiler.ts:2830]: I agree here but the toggles.ts are generic though in this case, they don't necessarily have to be related to the filters.

Comment (mattgodbolt) [reply static/panes/compiler.ts:2830]: My TS knowledge isn't strong enough to "template" the Toggles, but I'll try ...

Comment (mattgodbolt) [reply static/panes/compiler.ts:2830]: per discussion in Discord I'm stepping back from this until I understand TS more! Coming soon™

**Review by mattgodbolt — APPROVED** [review]
LGTM! will merge as this is clearly a step in the right direction. I've been testing and I can't see anything else broken

**Comment by mattgodbolt** [comment]
spoke too soon: choosing the template "diff" I get:
```
Internal Compiler Explorer error: TypeError: Cannot read properties of null (reading 'fullModule')
    at ClangCompiler.generateLLVMOptPipeline (/home/mgodbolt/dev/personal/ce/compiler-explorer/lib/base-compiler.ts:1049:44)
    at ClangCompiler.doCompilation (/home/mgodbolt/dev/personal/ce/compiler-explorer/lib/base-compiler.ts:1704:24)
    at async /home/mgodbolt/dev/personal/ce/compiler-explorer/lib/base-compiler.ts:2160:41
    at async run (/home/mgodbolt/dev/personal/ce/compiler-explorer/node_modules/p-queue/dist/index.js:163:29)
Compiler returned: -1```

looking

**Comment by mattgodbolt** [comment]
Looking good now, will merge once the tests are green

**Pull request merged**